src/json_merger.py

```diff
@@ -164,7 +166,7 @@ class NoneJsonMerger(BaseJsonMerger):
         json_obj: The string to merge into. (Should be 'None')
     """
 
-    json_obj: NoneType
+    json_obj: list | int | dict | str | NoneType
```

Comment:
Is this correct? the json obj should always be `NoneType`?

Reply:
you're allowing the value of json_obj to be set to all of those possible types in the below merge_ methods so the attribute should reflect those possibilities

Reply:
Oh I see, my intention here was to restrict it's typing when initialising the class. You're saying because we change it's type in subsequent methods that the type hinting should adhere to that too?

Reply:
That will work for the class instantiation, but if you later need to access that attribute the type hinting might not be accurate. If json_obj is private to the class, you could change this from a data class and just initialise a private _json_obj attribute in it's init as None